### PR TITLE
STORM-966 ConfigValidation.DoubleValidator is misleading and not neccessary

### PR DESCRIPTION
**Pull request by jerrypeng**
ConfigValidation.DoubleValidator doesn't really validate whether the type of the object is a double.

ConfigValidation.DoubleValidator code only checks if the object is null whether if the object is a instance of Number which is a parent class of Double.
DoubleValidator is only used once in Config.java and in that instance:
public static final Object TOPOLOGY_STATS_SAMPLE_RATE_SCHEMA = ConfigValidation.DoubleValidator;
can just be set to:
public static final Object TOPOLOGY_STATS_SAMPLE_RATE_SCHEMA = NUMBER.class;
Then we can just get rid of the misleading function ConfigValidation.DoubleValidator since it doesn't really check if a object is of double type thus the validator function doesn't really do anything and the name is misleading. In previous commit https://github.com/apache/storm/commit/214ee7454548b884c591991b1faea770d1478cec Number.Class was used anyway

### CODE REVIEW

**Comment by caofangkun** [comment]
+1

**Comment by HeartSaVioR** [comment]
You're right about DoubleValidator. DoubleValidator is not properly implemented.

Btw, Number.class seems not the proper thing to rely, too.
http://docs.oracle.com/javase/7/docs/api/java/lang/Number.html

BigInteger and BigDecimal extends Number, so if we don't want to handle big numbers in configuration, we should have properly implemented DoubleValidator (It should check the value is not beyond the range of Double), and don't use Number.class to validate.

**Comment by HeartSaVioR** [comment]
@jerrypeng 
For now I can make new issue to address it, and just vote +1 to this PR if you want.
Let me hear your thought about this. :)

**Comment by jerrypeng** [comment]
Perhaps for the purposes of the config "topology.stats.sample.rate" and future number based configs we should have a PositiveNumberValidator.  There is already a PositiveIntegerValidator but it only deals with positive integers.  It seems to me that "topology.stats.sample.rate" can be set to some positive float/double or integer. So I think we should include the PositiveNumberValidator.  I have added a commit that includes the code for the PositiveNumberValidator.  What do you think?

**Comment by HeartSaVioR** [comment]
@jerrypeng 
Great. It couldn't assert it is not big number or big decimal, but it's better than current.
+1.

**Comment by HeartSaVioR** [comment]
@jerrypeng Sorry, but could you provide unit test for PositiveNumberValidator? Thanks!

**Comment by jerrypeng** [comment]
added unit tests.  Please merge? or do I need to squash some of the commits?

File: storm-core/src/jvm/backtype/storm/ConfigValidation.java [hunk 1]
```diff
@@ -28,7 +28,6 @@
     /**
      * Declares methods for validating configuration values.
      */
-    public static interface FieldValidator {
```
Comment (caofangkun) [inline storm-core/src/jvm/backtype/storm/ConfigValidation.java]: This interface should not be annotated.
Could you please have a check?

**Comment by HeartSaVioR** [comment]
@jerrypeng There's a compile failure. Could you check it?

**Comment by jerrypeng** [comment]
You are right @caofangkun, not sure how that line got deleted, but it shouldn't have been removed.  I added back and I think the tests will pass now

**Comment by jerrypeng** [comment]
I don't know why the build is still failing. I just built and ran all the tests fine on my local machine.  I heard that since we just recently started using travis-ci that there are some bugs with it.

**Comment by caofangkun** [comment]
I checked the CI build log and find 
https://travis-ci.org/apache/storm/jobs/74073153#L4132

and there's a PR  on this , @HeartSaVioR Could you please review #652 ?

**Comment by HeartSaVioR** [comment]
+1. Build failure is unrelated. I did test before merge.

**Pull request merged**